Converted remaining icons to svg

bertyhell opened the pull request.

LordNyriox commented:
@bertyhell:  Out of curiosity, why did you change `checked.svg`?

Chocobo1 commented:
I found one icon in png format: src\icons\qbt-theme\rss-config
Do you mind convert it?

Also I can make the changes to make qbt pick them up, after the designs are finished.

Piccirello commented:
In some cases the WebUI still uses the png version (`kt-magnet.png`, `kt-set-max-download-speed.png`, `kt-set-max-upload-speed.png`, `slow_off.png`, etc). Are you going to update the appropriate html/js files, or should we tackle that in a separate PR?

Chocobo1 commented:
>Maybe we'd better replace it with application/rss+xml

application/rss+xml is already used as fallback, or you mean something else?

zeule commented:
Yes, I'd be happy to decrease the number of unique icons.

bertyhell commented:
> Out of curiosity, why did you change `checked.svg`?

Oeps, accidental override. Should be fixed now.

> Do you mind convert it?

Done

Piccirello commented:
What's your process for creating these svgs? In the future we may want to add additional icons.

Chocobo1 commented:
>>Do you mind convert it?
>
>Done

Thanks! Now I've pushed the changes for the code side.
Screenshot: ![screenshot](https://user-images.githubusercontent.com/9395168/45798356-3bfc0800-bcdc-11e8-8395-661683058c52.png)

Also I removed url.svg, seems it is not being used anywhere.

Chocobo1 commented:
@bertyhell 
I'm having problems with collapse-expand.svg, I suggest to split the icon into 2: collapse.svg & expand.svg.

bertyhell commented:
> I suggest to split the icon into 2: collapse.svg & expand.svg.

done

> What's your process for creating these svgs?

* Find a similar icon on [fontawesome](https://fontawesome.com/icons)
* If an identical exists use that
* If no icon exists create one or modify an existing one to be similar
* Replace its color with one of the main colors from the other icons (blue, orange, red, ...)
* Use the same artboard size as the existing icons
* Store icon using high accuracy (5 decimal points) svg v1

I'm using [illustrator](https://www.adobe.com/illustrator) to do this, but i guess [Inkscape](https://inkscape.org) would work as well.

Chocobo1 commented:
Update: removed expand.svg and rotate collapse.svg in css.

If there are no further issues, I'll cleanup the commits then it is ready for approval.

LordNyriox commented:
@Chocobo1:  Please do not forget to run `SVGO` on the icons, using the settings used by @sledgehammer999 in previous PRs.

Chocobo1 commented:
>@Chocobo1: Please do not forget to run SVGO on the icons, using the settings used by 
@sledgehammer999 in previous PRs.

OK, I'll remember that.

Chocobo1 commented on src/webui/www/private/index.html:
```diff
@@ -159,7 +159,7 @@ <h1 class="applicationTitle">qBittorrent Web User Interface <span class="version
                     <td><img id="connectionStatus" alt="Connection Status" src="images/skin/firewalled.svg" style="height: 1.5em;" /></td>
                     <td class="statusBarSeparator"></td>
-                    <td style="cursor:pointer;"><img id="alternativeSpeedLimits" alt="QBT_TR(Alternative speed limits)QBT_TR[CONTEXT=MainWindow]" src="images/slow_off.png" /></td>
+                    <td style="cursor:pointer;"><img id="alternativeSpeedLimits" alt="QBT_TR(Alternative speed limits)QBT_TR[CONTEXT=MainWindow]" src="images/slow_off.svg" /></td>
```
Thanks, fixed now.

Chocobo1 commented:
Update: cleanup commits, applied svgo.

sledgehammer999 commented:
Some icons have big empty canvas height:
* completed.svg
* downloading.svg
* slow.svg
* slow_off.svg
@bertyhell is there a reason for this?


Some icons are drawn with wrong color on Windows with Qt 5.11.1. Those are:
* slow.svg
* slow_off.svg
* rss-config.svg
* kt-set-max-upload-speed.svg
* kt-set-max-download-speed.svg

All those are drawn with black color. The WebUI draws them correctly though (firefox 62.0.2).
The commit without SVGO doesn't have this problem.

bertyhell commented:
> * is there a reason for this?

No, i fixed them.

> The commit without SVGO doesn't have this problem.

Reverted the svgo optimizations. If svgo causes any issues, we shouldn't apply it at all.
All optimized icons together are 112kb. So why would we even bother making that smaller? They are already much smaller than their png counterparts. And this is an installed program, not a website that has to be transfered every time.

sledgehammer999 commented:
>No, i fixed them.

completed.svg still renders as tall in the github file webview
downloading.svg renders as wide in the github file webview
slow_off.svg still renders as tall in the github file webview

Chocobo1 commented:
@sledgehammer999 
It seems further updates bring back the merge commits, can you squash them when you are ready to merge? I won't have enough time to redo it all over again. 
Good thing I saved the previous ready state in my repo: https://github.com/Chocobo1/qBittorrent/tree/pr_9499
You could use it as a base and merge fixups onto it.

DRSDavidSoft commented:
@bertyhell Thanks for _(slowly)_ changing the icon set, your design is **much** better than the old raster one.

Riotous commented:
Please fix all these icons, @sledgehammer999

sledgehammer999 commented:
I cleaned up the commit history and posted #9499 
Anyone interested please head over to the new PR. Ideally it should be approved and merged shortly.

Riotous commented:
@bertyhell Have you considered adding a "Check for updates" icon? That's the only one I see missing from 4.1.5 and this PR.

bertyhell commented:
> Have you considered adding a "Check for updates" icon? That's the only one I see missing from 4.1.5 and this PR.

@Riotous I'm not sure which icon you mean. For me, the "check for updates" menu option doesn't have an icon.
![image](https://user-images.githubusercontent.com/1710840/50663269-5a46f480-0fa9-11e9-81ef-0b3d2f5d3980.png)

Riotous commented:
@sledgehammer999 There's no need for an update icon but it would be nice to have one for the sake of making menus more consistent and giving them visual cues instead of just text. 

@bertyhell Yes, that's the one I mean. I opened an issue for this so you can add your icon when somebody decides to link an icon to that menu entry. 

https://github.com/qbittorrent/qBittorrent/issues/10110
